src/overlays/actors/ovl_En_Mb/z_en_mb.c

```diff
@@ -21,99 +20,1417 @@ const ActorInit En_Mb_InitVars = {
+            return;
+        case 1:
+            SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06014190, &D_0600EBE4, &this->limbDrawTable,
+                             &this->transitionDrawTable, 28);
+            this->actor.colChkInfo.health = 6;
+            this->actor.colChkInfo.mass = 0xFF;
+            this->actor.colChkInfo.damageTable = &sDamageTable2;
+            Actor_SetScale(&this->actor, 0.02f);
+            this->collider1.dim.height = 0xAA;
+            this->collider1.dim.radius = 0x2D;
+            this->actor.uncullZoneForward = 4000.0f;
+            this->actor.uncullZoneScale = 800.0f;
+            this->actor.uncullZoneDownward = 1800.0f;
+            this->unk_364 = 710.0f;
+            this->collider2.body.toucher.flags = 0x20000000;
+            temp_a0_2 = (this->actor.posRot.rot.y - Math_Vec3f_Yaw(&this->actor.posRot.pos, &player->actor.posRot.pos));
+            if (temp_a0_2 >= 0) {
+                phi_v1 = temp_a0_2;
+            } else {
+                phi_v1 = 0 - temp_a0_2;
+            }
+            if (phi_v1 >= 0x4001) {
+                this->actor.posRot.rot.y = thisx->posRot.rot.y + 0x8000;
+                this->actor.shape.rot.y = thisx->posRot.rot.y;
+                this->actor.posRot.pos.z = thisx->posRot.pos.z + 600.0f;
+            }
+            ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawFunc_Teardrop, 90.0f);
+            this->actor.flags &= -2;
+            this->actor.naviEnemyId++;
+            func_80AA6898(this);
+            return;
+    }
+
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06008F38, &D_060028E0, &this->limbDrawTable,
+                     &this->transitionDrawTable, 28);
+    Actor_SetScale(&this->actor, 0.014f);
+    this->unk_35D = (thisx->params & 0xFF00) >> 8;
+    this->actor.params = 1;
+    this->unk_35C = 0;
+    this->actor.colChkInfo.health = 1;
+    this->actor.colChkInfo.mass = 0xFE;
+    this->unk_360 = 350.0f;
+    this->unk_364 = 1750.0f;
+    this->actor.flags &= -2;
+    func_80AA68FC(this, globalCtx);
+}
+
+void EnMb_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnMb* this = THIS;
+    Collider_DestroyTris(globalCtx, &this->collider3);
+    Collider_DestroyCylinder(globalCtx, &this->collider1);
+    Collider_DestroyQuad(globalCtx, &this->collider2);
+}
+
+void func_80AA6408(EnMb* this, GlobalContext* globalCtx) {
+    s16 yaw;
+
+    yaw = Math_Vec3f_Yaw(&this->actor.posRot.pos, &this->unk_344);
+    this->actor.shape.rot.y = yaw;
+    this->actor.posRot.rot.y = yaw;
+}
+
+#ifdef NON_MATCHING
+// Very small regaloc
+void func_80AA6444(EnMb* this, GlobalContext* globalCtx) {
+    Path* path;
+    Vec3s* pathPos;
+    s8 temp;
+
+    path = &globalCtx->setupPathList[this->unk_35D];
+    if (this->unk_35C == 0) {
+        this->unk_35E = 1;
+    } else {
+        temp = (path->count - 1) & 0xFFFFFFFFFFFFFFFF;
+        if (temp == this->unk_35C) {
+            this->unk_35E = -1;
+        }
```

Comment:
```suggestion

    path = &globalCtx->setupPathList[this->unk_35D];
    if (this->unk_35C == 0) {
        this->unk_35E = 1;
    } else if (this->unk_35C == (path->count - 1))  {
        this->unk_35E = -1;
    }
```
The comparison needed to be switched around.

Reply:
I tried it like that but it didnt work for me.
![func_80AA6444](https://user-images.githubusercontent.com/35883445/96375010-60896f00-1144-11eb-8cb8-33213d589954.PNG)

Reply:
Oh whoops I meant `(s8)(path->count - 1)`